src/dil_demo/wms/verify.clj

```diff
@@ -10,52 +10,73 @@
             [dil-demo.ishare.policies :as policies]
             [dil-demo.otm :as otm]))
 
-(defn rejection-reasons [client-data {:keys [issuer target mask]}]
-  (try
-    (-> client-data
-        (assoc :ishare/policy-issuer issuer ;; ensures we target the right AR
-               :ishare/message-type :delegation
-               :ishare/params mask)
-        ishare-client/exec
-        :ishare/result
-        :delegationEvidence
-        (policies/rejection-reasons target))
-    (catch Throwable ex
-      (prn ex)
-      [(str "Technische fout opgetreden: " (.getMessage ex))])))
+(defn ishare-delegation! [{:keys [client-data] :as req}
+                          title
+                          {:keys [issuer target mask]}]
+  (binding [ishare-client/log-interceptor-atom (atom [])]
+    (try
+      (-> req
+          (update :delegation-evidences (fnil conj [])
+                  [target
+                   (-> client-data
+                       (assoc :ishare/policy-issuer issuer ;; ensures we target the right AR
+                              :ishare/message-type :delegation
+                              :ishare/params mask)
+                       (ishare-client/exec)
+                       :ishare/result
+                       :delegationEvidence)])
+          (update-in [:explanation] (fnil conj [])
+                     [title @ishare-client/log-interceptor-atom]))
+      (catch Throwable ex
+        (-> req
+            (update :delegation-evidences (fnil conj [])
+                    [target {}])
+            (update-in [:explanation] (fnil into [])
+                       [[title @ishare-client/log-interceptor-atom]
+                        [(str "Technische fout upgrade: " (.getMessage ex))]]))))))
+
+(defn rejection-reasons [{:keys [delegation-evidences]}]
+    (seq (mapcat (fn [[target delegation-evidence]]
+                   (policies/rejection-reasons delegation-evidence target))
+                 delegation-evidences)))
+
+(defn permitted? [req]
+  (not (rejection-reasons req)))
 
-(defn verify-owner
+(defn verify-owner!
```

Comment:
doet dit een update ergens in een AR of heeft deze functie een `!` omdat het een remote query doet?